qt/qt_common/proxy_style.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <QtWidgets/QStyle>
+
+namespace qt
+{
+namespace common
+{
+class ProxyStyle : public QStyle
+{
+public:
+  explicit ProxyStyle(QStyle * p);
+
+  void drawComplexControl(ComplexControl control, const QStyleOptionComplex * option,
+                          QPainter * painter, const QWidget * widget = 0) const;
```

Comment:
Moreover, this requires a lot of changes, non-relevant to the PR. How about to do this later?